pack/set.go

```diff
@@ -60,6 +60,12 @@ func NewSet(db string) (*Set, error) {
 			// We have a pack (since it matched the regex), but the
 			// index is missing or unusable.  Skip this pack and
 			// continue on with the next one, as Git does.
+			if idxf != nil {
```

Comment:
> Having said that, I'm not opposed to being careful here if we think that might change in the future.

This is what motivated that change here (though I should have mentioned so explicitly in my original patch, and I am sure that there are many places throughout the code that _don't_ do this).

I agree with your reading of the implementation, but I am hesitant to drop this patch based on that alone. Since the documentation doesn't promise anything about the relation between the `*os.File` and `error` returned, I don't think that we can rely on the source alone.